### PR TITLE
Support configuration files

### LINKED ISSUE #233: Support configuration files
### Current state
All options are given over the CLI which works fine for the most use cases

### My suggestion
In some use cases it's better to use configuration files to define options. In addition, these options can be overwritten with options given to the CLI.

### Implementation
Personally, I would first create a configuration file reader (abstract so it can handle other files in the future). Then I would add a TOML parser (pyproject.toml), because this seems to be the go-to way.

Example toml
```toml
[tool.diff_cover]
compare_branch = "origin/foo"
fail_under = 42

[tool.diff_quality]
fail_under = 23
include = [
  "project/foo/**"
]
```
So, the toml file:

- uses both tools with their name
- same options as on CLI, but with underscores
- if an option can be specified multiple times, the value is a list

### Flow

1. Check if the user specifies a configuration file over CLI; if yes, use that (`--config myfile.toml`)
2. If no, then check if a pyproject.toml exists and check if it contains any section which belongs to us **(do we want to have a magic file name?)**
3. Afterwards merge all configuration values with the given CLI parameters (CLI overwrite config)
4. Proceed as normal

### PR DESCRIPTION
**Pull request by kasium**
Resolves #233

### CODE REVIEW

**Comment by kasium** [comment]
Here the first draft without tests. Please feel free to comment. Afterwards I'll add the tests

File: pyproject.toml [hunk 1]
```diff
@@ -62,6 +62,9 @@ black = "^21.7b0"
 isort = "^5.9.3"
 doc8 = "0.9.0"
 
+[tool.poetry.extras]
+toml = ["tomli"]
+
```
Comment (Bachmann1234) [inline pyproject.toml:67]: Wait, whats the difference between what you do here and what I setup in line 48 where I specify this as optional...

oh.. is this what allows `pip install diff_cover[tomli]` ?  In that case I would just say `toml` here. In the off chance we change how the toml stuff works down the line I would not want to have to change this.

**Comment by Bachmann1234** [comment]
This looks pretty good to me. I left one comment . Looking forward to the tests but the broad strokes look right

File: pyproject.toml (hunk 1 above)
Comment (kasium) [reply pyproject.toml:67]: Exactly, you got it. `tomli = {version = "^1.2.1", optional = true}` is just the pure dependency. Then `toml = ["tomli"]` declares `tomli` has an extra dependency which can be installed with `pip install diff_cover[toml]` (not tomli)

**Comment by Bachmann1234** [comment]
@kasium this is out https://pypi.org/project/diff-cover/6.3.0/ thanks again!

**Pull request merged**